src/seq.rs

```diff
@@ -601,23 +654,41 @@ impl fmt::Debug for Event {
     }
 }
 
-/// Low level methods to set/get data on an Event. Don't use these directly, use generic methods on Event instead.
-pub trait EventData {
+/// Low level method to check if event type supports a given data type. Don't use directly, use generic methods on Event instead.
+pub trait HasEventData {
```

Comment:
Yet another comment about this: My changes allowed creation of a sysex event from a `Vec<u8>` by move, so the `Vec` doesn't need to be copied. This is not possible with the original design, because `Event::new` takes `&D`, so it only has a reference of the vector and needs to copy it. I think the best way forward here is the following: 
Create a constructor especially for event types with variable length data (e.g. `Event::new_varlen`) that takes a `T: Into<Cow<'a, [u8]>>`. That way you can pass either a slice or a vector (by value).
A corresponding `get_data_varlen` should return `&'a [u8]`. What do you think?